pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -54,8 +54,8 @@ func (e *ErrorWithCodes) Error() string {
 
 var (
 	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|InvalidAuthenticationTokenTenant|SignatureDoesNotMatch|Authentication failed|AuthFailure|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId)`)
-	quotaExceededRegexp                 = regexp.MustCompile(`(?i)(LimitExceeded|Quotas|Quota exceeded|exceeded quota|Quota has been met)`)
-	rateLimitsExceededRegexp            = regexp.MustCompile(`(?i)(Throttling|Too many requests)`)
+	quotaExceededRegexp                 = regexp.MustCompile(`(?i)([^Request]LimitExceeded|Quotas|Quota exceeded|exceeded quota|Quota has been met)`)
+	rateLimitsExceededRegexp            = regexp.MustCompile(`(?i)(RequestLimitExceeded|Throttling|Too many requests)`)
```

Comment:
Can you add a test about this?

Reply:
We only have basic tests for these error mappings and decided not to pollute them with the many hundred combinations of potential error messages that are expected to be mapped correctly. Hence, yes, I could do it, but I don't see why this should be special now, so I'd rather refrain from doing this (unless we do it consistently for all the error expressions).

Reply:
I think one line test here https://github.com/gardener/gardener/blob/b43d545d1957318dee1a17116707251ca7fef405/pkg/apis/core/v1beta1/helper/errors_test.go#L38-L63 will be useful to verify that `RequestLimitExceeded` error is not detected as quota exceeded error.